lib/elixir/lib/kernel.ex

```diff
@@ -5135,9 +5135,8 @@ defmodule Kernel do
     String.to_charlist(:elixir_interpolation.unescape_chars(string))
   end
 
-  defmacro sigil_c({:<<>>, meta, pieces}, []) do
-    binary = {:<<>>, meta, unescape_tokens(pieces)}
-    quote(do: String.to_charlist(unquote(binary)))
+  defmacro sigil_c({:<<>>, _meta, pieces}, []) do
+    quote(do: List.to_charlist([unquote_splicing(unescape_list_tokens(pieces))]))
```

Comment:
Is there a difference between `[unquote_splicing(list)]` and `unquote(list)`?

Reply:
There shouldn't. @ericmj good catch!

Reply:
There's not. It's a left over from a different implementation and some refactoring.